portal/auth.py

```diff
@@ -0,0 +1,64 @@
+import functools
+
+from flask import (
+    Blueprint, flash, g, redirect, render_template, request, session, url_for
+)
+from werkzeug.security import check_password_hash, generate_password_hash
+
+from . import db
+
+bp = Blueprint('auth', __name__, url_prefix='/auth')
+
+@bp.route('/login', methods=('GET', 'POST'))
+def login():
+    email = request.args.get('email')
+    password = request.args.get('password')
+    error = None
+    cur = db.get_db().cursor()
+    cur.execute(
+        'SELECT * FROM users WHERE email = %s', (email,)
+    )
+    user = cur.fetchone()
+
+    if user is None:
+        error = 'Incorrect email.'
+    elif not user['password']:
+        error = 'Incorrect password.'
```

Comment:
Make sure you're following OWASP security guidelines. We don't want to reveal too much info to attackers.